### PR TITLE
Parameterises container registry and pull policy with defaults

### PR DESCRIPTION
**Pull request by cmgrote**
To address #739

### CODE REVIEW

File: open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml [hunk 1]
```diff
@@ -55,71 +55,61 @@ debug:
 logging:
   level: NONE
 
-
-# The following section defines all of the DOCKER images being used by this
-# chart. Normally they should be left as is, but are exposed here in case the
-# user wishes to extend.
+imageDefaults:
+  registry: "localhost:5000"
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml]: I am considering reverting the default registry to dockerhub (and making the repository configurable).
I am still awaiting the LF build team, but in the meantime I've had a couple of people ask about using the charts.
The easiest way of doing that is to ensure the images work 'out of the box' MUCH though I dislike the non-official repo id which I will again try and push forward. For them to build, and setup a repo, is not as easy - though it is important to have the capability

I can do this under an additional PR if easier - we don't need to change this one /miss out on the other changes

File: open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml [hunk 2]
```diff
@@ -55,71 +55,61 @@ debug:
...
+    registry: "docker.io/library"
     repository: mariadb
-    tag: latest
-    pullPolicy: Always
 
   egeria:
-    repository: localhost:5000/odpi/egeria
-    tag: 1.1-SNAPSHOT
-    pullPolicy: Always
+    repository: odpi/egeria
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml]: the repo name needs to be configurable (see above comment) - clearly it already is, but we can default it too, for the egeria custom images (ie not mariadb)

File: open-metadata-resources/open-metadata-deployment/docker/configure/pom.xml [hunk 3]
```diff
@@ -32,17 +32,45 @@
                 <artifactId>dockerfile-maven-plugin</artifactId>
```
Comment (planetf1) [inline open-metadata-resources/open-metadata-deployment/docker/configure/pom.xml]: I think this should work - the two step process is needed to get the latest approach working.
I can't test in the next few hours as the full build is too long, so will raise a new issue if any issues observed

**Comment by planetf1** [comment]
See review above. Despite review comment merging now is probably the best approach, and a new PR can cater to the other changes mentioned?

File: open-metadata-resources/open-metadata-deployment/docker/configure/pom.xml (hunk 3 above)
Comment (cmgrote) [reply open-metadata-resources/open-metadata-deployment/docker/configure/pom.xml]: I tested on my side and it worked, but would indeed be good to have a peer-reviewed test =)

File: open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml (hunk 1 above)
Comment (cmgrote) [reply open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml]: Raised #829

File: open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml (hunk 2 above)
Comment (cmgrote) [reply open-metadata-resources/open-metadata-deployment/charts/vdc/values.yaml]: Included in #829

**Pull request merged**